src/select.jsx

```diff
@@ -72,30 +72,41 @@ var classBase = React.createClass({
     };
   },
   getInitialState () {
+    var initialIndex = this.getValueIndex(this.props.defaultValue);
```

Comment:
If `this.props.defaultValue` is `undefined` you're still looping through all the children in `getValueIndex`